spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkV2Filters.java

```diff
@@ -360,10 +417,70 @@ private static boolean hasNoInFilter(Predicate predicate) {
   }
 
   private static boolean isSupportedInPredicate(Predicate predicate) {
-    if (!isRef(childAtIndex(predicate, 0))) {
+    if (!canConvertToTerm(childAtIndex(predicate, 0))) {
       return false;
     } else {
       return Arrays.stream(predicate.children()).skip(1).allMatch(SparkV2Filters::isLiteral);
     }
   }
+
+  /** Should be called after {@link #canConvertToTerm} passed */
+  private static <T> UnboundTerm<Object> toTerm(T input) {
+    if (input instanceof NamedReference) {
+      return Expressions.ref(SparkUtil.toColumnName((NamedReference) input));
+    } else {
+      return udfToTerm((UserDefinedScalarFunc) input);
+    }
+  }
+
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  private static UnboundTerm<Object> udfToTerm(UserDefinedScalarFunc udf) {
+    switch (udf.name().toLowerCase(Locale.ROOT)) {
+      case "years":
+        if (udf.children().length == 1 && isRef(udf.children()[0])) {
```

Comment:
Minor: there's quite a bit of duplication across the simple transform cases. It would be nice to refactor this:

```java
  Expression[] children = udf.children();
  if (children.length == 1) {
    Expression child = udf.children()[0];
    if (isRef(child)) {
      String column = SparkUtil.toColumnName((NamedReference) child);
      switch (udf.name().toLowerCase(Locale.ROOT)) {
        case "years":
          return year(column);
        case "months":
          return months(column);
        ...
      }
    }
  } else if (children.length == 2) {
    ...
  }
```

Reply:
Updated.